src/PoolManager.sol

```diff
@@ -185,11 +185,12 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
             })
         );
 
-        _accountPoolBalanceDelta(key, delta);
-
         emit ModifyLiquidity(id, msg.sender, params.tickLower, params.tickUpper, params.liquidityDelta);
 
-        key.hooks.afterModifyLiquidity(key, params, delta, hookData);
+        BalanceDelta hookDelta = key.hooks.afterModifyLiquidity(key, params, delta, hookData);
```

Comment:
This doesn't seem to be supporting custom accounting. the LP's delta is being supplied by the v3 curve math instead of by the hook (maybe it's a V2 hook). How does the hook communicate to the LP, "transfer this much to the pool which should be available to me, the hook"

Reply:
This could be achieved by passing in a liquidityDelta of 0 and packing the actual custom liquidity information in the hookData, or just interacting directing with the hook